### DISCUSSION TITLE
How do i check if a Chest is part of a doublechest?

### DISCUSSION THREAD

**Discussion by CHECKERCE** [PrismarineJS/mineflayer#3139] [General]
my approach would be to look at the chest's NBT Data which looks something like this:

`[facing=south,type=right,waterlogged=false]{Items:[{Count:64b,Slot:0b,id:"minecraft:crimson_stem"}]}`

`type=right` indicates that it is the right side of a double chest,
but i have no idea of how to get the NBT Data of a block entity.

I tried:
`let chestBlock = bot.blockAt(chestPosition, true)`
`blockEntity = chestBlock.blockEntity`

because the documentation states: 
> Also block.blockEntity is additional field with block entity data as Object.

but this doesn't work, blockEntity is null

**Comment by amoraschi** [comment]
You could try checking the `block.shapes`, which is an Array of numbers that indicate the dimensions of the block

**Reply by CHECKERCE** [reply]
tried it and it actually works.
it's a weird approach but i will use it until i find a better option.

thank you.
